src/Adapter/Product/Update/ProductShopUpdater.php

```diff
@@ -0,0 +1,302 @@
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\Update;
+
+use DateTime;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Query\QueryBuilder;
+use PrestaShop\PrestaShop\Adapter\Product\Repository\ProductRepository;
+use PrestaShop\PrestaShop\Adapter\Shop\Repository\ShopRepository;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopId;
+use PrestaShop\PrestaShop\Core\Util\DateTime\DateTime as DateTimeUtil;
+use Product;
+
+class ProductShopUpdater
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @var string
+     */
+    private $dbPrefix;
+
+    /**
+     * @var ProductRepository
+     */
+    private $productRepository;
+
+    /**
+     * @var ShopRepository
+     */
+    private $shopRepository;
+
+    /**
+     * @param Connection $connection
+     * @param string $dbPrefix
+     * @param ProductRepository $productRepository
+     * @param ShopRepository $shopRepository
+     */
+    public function __construct(
+        Connection $connection,
+        string $dbPrefix,
+        ProductRepository $productRepository,
+        ShopRepository $shopRepository
+    ) {
+        $this->connection = $connection;
+        $this->dbPrefix = $dbPrefix;
+        $this->productRepository = $productRepository;
+        $this->shopRepository = $shopRepository;
+    }
+
+    /**
+     * @param ProductId $productId
+     * @param ShopId $sourceShopId
+     * @param ShopId $targetShopId
+     */
+    public function copyToShop(ProductId $productId, ShopId $sourceShopId, ShopId $targetShopId): void
+    {
+        $this->shopRepository->assertShopExists($sourceShopId);
+        $this->shopRepository->assertShopExists($targetShopId);
+
+        /** @var Product $sourceProduct */
+        $sourceProduct = $this->productRepository->getForShop($productId, $sourceShopId);
+
+        // @todo: for now only fields from product_shop table ar handed, we still need to handle multilang, stock_available, and probably other things (in another PR)
+        // @todo: do not forget to copy customization_field_lang, this part was not handled in the legacy import (fixed in this PR)
+        // The fields are fetched separately for more clarity, and it could also allow to configure which parts are copied
+        // (e.g copy only prices but not stock)
+        $fields = array_merge(
```

Comment:
In case of arrays are in multidimensional mode: https://www.php.net/manual/en/function.array-merge-recursive.php

Reply:
For now only the "simple" fields directly related to Product are handled, so they are only single dimension arrays, it's fine with the current code

The more advanced fields (like relationships) will be handled in a different PR, and probably in a different function (well another private function at least)